lib/features/modeling/ElementFactory.js

```diff
@@ -101,7 +101,9 @@ ElementFactory.prototype.createBpmnElement = function(elementType, attrs) {
 
   if (attrs.eventDefinitionType) {
     eventDefinitions = businessObject.get('eventDefinitions') || [];
-    newEventDefinition = this._moddle.create(attrs.eventDefinitionType);
+    newEventDefinition = this._moddle.create(attrs.eventDefinitionType, {
```

Comment:
The right place to encode the _always needs ID constraint` would be [`BpmnFactory#L16`](https://github.com/bpmn-io/bpmn-js/blob/master/lib/features/modeling/BpmnFactory.js).